core/src/main/java/org/apache/spark/io/NioBufferedFileInputStream.java

```diff
@@ -66,32 +66,30 @@ private boolean refill() throws IOException {
         nRead = fileChannel.read(byteBuffer);
       }
       byteBuffer.flip();
-      if (nRead < 0) {
-        return false;
-      }
+      return nRead >= 0;
     }
     return true;
   }
 
   @Override
   public synchronized int read() throws IOException {
-    if (!refill()) {
-      return -1;
+    if (refill()) {
```

Comment:
@mridulm I agree with your viewpoint. Currently, I have transformed the code from checking `!refill()` to checking `refill()`, and adjusted the corresponding handling logic (I have not yet updated the PR description). Do you consider this to be acceptable? If not, I will close this PR.